docs/developer-docs/latest/developer-resources/plugin-api-reference/server.md

```diff
@@ -0,0 +1,464 @@
+| `default`   | Object, or Function that returns an Object | Default plugin configuration, merged with the user configuration                                                                                         |
+| `validator` | Function                                       | <ul><li>Checks if the results of merging the default plugin configuration with the user configuration is valid</li><li>Throws errors when the resulting configuration is invalid</li></ul> |
+
+**Example:**
+
+```js
+// path: ./strapi-server.js
+
+const config = require('./config');
+
+module.exports = () => ({
+  config: {
+    default: ({ env }) => { optionA: true },
+    validator: (config) => { 
+      if (typeof config.optionA !== boolean) {
+        throw new Error('optionA has to be a boolean');
+      }
+    },
+  },
+});
+```
+
+## Backend customization
+
+### Content-Types
+
+An object with the [Content-Types](/developer-docs/latest/development/backend-customization.md#models) the plugin provides.
+<!-- TODO: update link to Backend customization > models once merged with database PR -->
+
+**Type**: `Object`
+
+<!-- ! The link to the `info` section below won't work in this PR, but will work once the content is merged with the database PR -->
+:::note
+Content-Types keys in the `contentTypes` object should re-use the `singularName` defined in the [`info`](/developer-docs/latest/development/backend-customization/models.md#model-information) key of the schema.
+:::
+
+**Example:**
+
+```js
+// path: ./strapi-server.js
+
+const contentTypes = require('./content-types');
+
+module.exports = () => ({
+  contentTypes,
+});
+```
+
+```js
+// path: ./content-types/index.js
+
+const contentTypeA = require('./content-type-a');
+const contentTypeB = require('./content-type-b');
+
+module.exports = [
+  contentTypeA, // should re-use the same
+  contentTypeB,
+];
+```
+
+```js
+// path: ./content-types/content-type-a.js
+
+module.exports = {
+  info: {
+    tableName: 'content-type',
+    singularName: 'contentTypeA', // camel case mandatory
+    pluralName: 'contentTypeAs', // camel case mandatory
+    displayName: 'Content Type A',
+    description: 'A regular content type'
+    kind: 'collectionType'
+  },
+  options: {
+    draftAndPublish: true,
+  },
+  pluginOptions: {
+    'content-manager': {
+      visible: false
+    },
+    'content-type-builder': {
+      visible: false
+    }
+  },
+  attributes: {
+    name: {
+      type: 'string',
+      min: 1,
+      max: 50,
+      configurable: false
+    },
+  }
+};
+```
+
+### Routes
+
+<!-- ? Have we decided on/implemented routes behavior yet? -->
+
+An array of [routes](/developer-docs/latest/development/backend-customization/routing.md) configuration.
+
+**Type**: `Object[]`
+
+**Example:**
+
+```js
+// path: ./strapi-server.js
+
+const routes = require('./routes');
+
+module.exports = () => ({
+  routes,
```

Comment:
Should we talk about the API type? @alexandrebodin 
```js
  type: 'content-api',
```

Reply:
@alexandrebodin Have we decided about the type implementation yet? (`content-api` or `admin-api`)